tools/slicec-cs/src/generators/enum_generator.rs

```diff
@@ -198,6 +466,18 @@ fn enum_encoder_extensions(enum_def: &Enum) -> CodeBlock {
     )
     .add_generated_remark("static class", enum_def);
 
+    let encode_body = if enum_def.is_mapped_to_cs_enum() {
+        format!(
+            "{encode_enum}(({cs_type})value);",
+            encode_enum = match &enum_def.underlying {
+                Some(underlying) => format!("encoder.Encode{}", underlying.definition().type_suffix()),
+                None => "encoder.EncodeSize".to_owned(),
+            },
+        )
+    } else {
+        "value.Encode(ref encoder);".into()
```

Comment:
When converting from a string literal to an owned `String`, `to_owned` is more idiomatic (and expressive) than `into`.
```suggestion
        "value.Encode(ref encoder);".to_owned()
```